cli/cmd/start/start.go

```diff
@@ -143,6 +143,14 @@ func StartCmd(ch *cmdutil.Helper) *cobra.Command {
 			}
 			localURL := fmt.Sprintf("%s://localhost:%d", scheme, httpPort)
 
+			// if olapDriver is clickhouse and no olapDSN is specified, run ClickHouse as a subprocess
+			if olapDriver == "clickhouse" && olapDSN == local.DefaultOLAPDSN {
+				olapDSN, err = startClickHouse(cmd.Context(), projectPath)
+				if err != nil {
+					return err
+				}
+			}
```

Comment:
Unfortunately this is not a reliable way of discovering if ClickHouse is the OLAP engine because these flags are deprecated: https://github.com/rilldata/rill/blob/b67474d7b4fd3770f21999f0c3e96a3e86c216b7/cli/cmd/start/start.go#L196-L199

We now support having one project connect to multiple OLAP engines simultaneously by configuring `connector: name` in the individual sources and models. To avoid repetition, a default engine can be configured in `rill.yaml` using `olap_connector: ...`.

I think the only way to handle this is to move the handling of starting the ClickHouse subprocess into `runtime/drivers/clickhouse` in the `driver.Open` function. This has some nice other advantages:
- Provides easy access to the `DataDir` and `TempDir` (they are automatically made available as connector properties [here](https://github.com/rilldata/rill/blob/b67474d7b4fd3770f21999f0c3e96a3e86c216b7/runtime/connections.go#L281-L285)).
- Allows overriding parameters like the port using variables, maybe something like `--var connector.clickhouse.embed_port`.
- Centralizes ClickHouse logic in the ClickHouse driver package.
- The `Close` function on the driver handle is automatically called when interrupt signals are received, so the subprocess doesn't need to listen for OS signals directly. It can just respond to `Close` calls.